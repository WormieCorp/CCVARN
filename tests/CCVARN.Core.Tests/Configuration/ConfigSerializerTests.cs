namespace CCVARN.Core.Tests
{
	using System.Collections.Generic;
	using System.IO;
	using ApprovalTests;
	using CCVARN.Core.Configuration;
	using NUnit.Framework;

	[NonParallelizable]
	public class ConfigSerializerTests
	{
		private static readonly string testDirectory = Path.Combine(Path.GetTempPath(), "ccvarn-tests");
		private static readonly string testFile = Path.Combine(testDirectory, "CCVARN.yml");

		[SetUp]
		public void CreateTestDirectory()
		{
			if (!Directory.Exists(testDirectory))
				Directory.CreateDirectory(testDirectory);
		}

		[OneTimeTearDown]
		public void RemoveTestFolder()
		{
			Directory.Delete(testDirectory, true);
		}

		[Test]
		public void SerializingDefaultConfiguration()
		{
			var config = new Config();
			ConfigSerializer.SaveConfiguration(testDirectory, config);

			Approvals.VerifyFile(testFile);
		}

		[Test]
		public void SerializingConfigurationWithCustomSettings()
		{
			var config = new Config
			{
				NextVersion = "0.5.0",
				Tag = "rc",
				TypeScopes = new HashSet<TypeScope>
				{
					new TypeScope("yes", new Description("Singular", "Plural"), "my scope"),
					new TypeScope("feature", new Description("Feature", "Features"))
				}
			};
			ConfigSerializer.SaveConfiguration(testDirectory, config);
			Approvals.VerifyFile(testFile);
		}
	}
}
