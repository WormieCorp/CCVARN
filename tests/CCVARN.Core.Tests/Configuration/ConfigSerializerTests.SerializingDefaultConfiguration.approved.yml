# The major-minor-patch version that should be used when parsing the commits.
# This option will only be taken into account when the current HEAD is not a tagged
# build, and the value is higher than the last tagged release.
next-version: 0.0.0
# The pre-release tag to use as a default for non-tagged releases (tagged releases will override this option).
tag: alpha
# The types and scope that should be consider valid, as well as which part of the version to bump when the type+scope matches.
# Additionally the headers in the changelog can be configured,
# and wether the types+scope should be included in the changelog by default
# (breaking changes will always be included in the changelog).
types:
- description:
    plural: Features
    singular: Feature
  scope: 
  type: feat
  bump: Minor
- description:
    plural: Bug Fixes
    singular: Bug fix
  scope: 
  type: fix
  bump: Patch
- description:
    singular: Performance Improvements
  scope: 
  type: perf
  bump: Minor
- description:
    plural: Improvements
    singular: Improvement
  scope: 
  type: improvement
  bump: Minor
- description:
    singular: Reverts
  scope: 
  type: revert
- description:
    singular: Documentation
  changelog: false
  scope: 
  type: docs
- description:
    plural: Styles
    singular: Style
  changelog: false
  scope: 
  type: style
- description:
    singular: Code Refactoring
  changelog: false
  scope: 
  type: refactor
- description:
    plural: Tests
    singular: Test
  changelog: false
  scope: 
  type: test
- description:
    singular: Chore
  changelog: false
  scope: 
  type: chore
- description:
    plural: Builds
    singular: Build
  changelog: false
  scope: 
  type: build
- description:
    singular: Continuous Integration
  changelog: false
  scope: 
  type: ci
